### ISSUE TITLE
No sound

### ISSUE THREAD

**Issue by TTimoNN** [PC-Logix/OpenFM#57]
Hi, I'm running 
OpenFM-1.7.10-0.1.0-44.jar

1. I spawn in the radio/speaker, place both down and link them with the radio tuner.

log:
[20:57:24] [Client thread/INFO]: [CHAT] [OpenFM] Selected Speaker
[20:57:25] [Client thread/INFO]: [CHAT] [OpenFM] Speaker Added

Then I add your (most-likely known-to-work) radio: http://listen.radionomy.com/DeepHouse

Then I hear nothing playing. Except for the stock minecraft music if that is the case at that time.

log:
[20:58:26] [Client thread/INFO] [openfm]: MP3
[20:58:26] [Client thread/INFO] [openfm]: Starting Stream: http://listen.radionomy.com/DeepHouse at X:663 Y:4 Z:220

Extra information:
I'm running Linux, which means that all minecraft audio is being outputted to PulseAudio.

All friends that run on Windows have it working.

So this can be narrowed down to what output device you have choosen in the mod. (I think, I don't know java so I couldn't check)

Thanks!

**Comment by CaitlynMainer** [comment]
Which linux distro are you on?  I've had one other report of issues on Arch, but it works fine for me in Debian, Ubuntu, and Fedora.

**Comment by TTimoNN** [comment]
Arch.

**Comment by CaitlynMainer** [comment]
No, they made a VM for me to try to use to reproduce it but it never booted, I've found VM images from osbox that have recent versions of arch that I'll try when I get off work.

**Comment by TTimoNN** [comment]
Awesome, if there is anything I can do in the mean time tell me.

I'm looking forward to your discoveries.

**Comment by CaitlynMainer** [comment]
I just tested on Arch with ALSA, and had a friend test with Pulse, and both of us were able to hear playback from OpenFM.  Just in case check your Jukebox/Noteblock volume settings for Minecraft.

**Comment by TTimoNN** [comment]
I did check all my in-game sound settings. And they're on.
I've tried running it on two different Arch installations on two different systems.
Is there anything I can do to increase the debugging in the logs?

**Comment by CaitlynMainer** [comment]
Could you please try this build in single player http://puu.sh/u4y4Z/23de2ef396.jar   this is the latest build for 1/9/1.10 with lots of logging in the playback methods, just try to play a stream, wait a minute or so, close the game and give me the entire log in a pastebin.

**Comment by LizzyTrickster** [comment]
@TTimoNN mind pasting the output of `pacman -Q | grep -i "alsa\|pulse"? It may be that you don't have a package that's needed.
I was the other user @CaitlynMainer said about, mine wasn't working a while back but it's working after I re-did my Arch install so I think i got an extra package that may have not been there earlier

**Comment by TTimoNN** [comment]
sudo pacman -Q | grep -e alsa -e pulse

alsa-lib 1.1.3-1
alsa-plugins 1.1.1-1
alsa-utils 1.1.3-1
apulse 0.1.7-1
lib32-alsa-lib 1.1.3-1
lib32-alsa-plugins 1.1.1-1
lib32-libpulse 10.0-1
libcanberra-pulse 0.30-8
libpulse 10.0-2
pulseaudio 10.0-2
pulseaudio-alsa 2-3
zita-alsa-pcmi 0.2.0-3

testing custom build, sec.

**Comment by LizzyTrickster** [comment]
hmm, I don't have alsa-utils, apulse or any of the 32bit libs but i do have pulseaudio-gconf though comparing my list with another working arch user's they don't have gconf so it's not that....
This just gets stranger and stranger....
What Java/JDK do you have?

**Comment by TTimoNN** [comment]
openjdk version "1.8.0_121"
OpenJDK Runtime Environment (build 1.8.0_121-b13)
OpenJDK 64-Bit Server VM (build 25.121-b13, mixed mode)

**Comment by CaitlynMainer** [comment]
Oh.. my bad for some reason I thought your ticket said 1.9... I can do this in 1.7 as well, it'll just be a bit

**Comment by TTimoNN** [comment]
That would be very much appreciated! I'll wait for your response.

**Comment by CaitlynMainer** [comment]
http://puu.sh/u4HuY/672ac52dde.jar  Hopefully this works I'm at work and none of these computers can run MC so this is untested.

**Comment by TTimoNN** [comment]
on it, will report back asap

**Comment by TTimoNN** [comment]
[21:07:40] [Client thread/INFO] [openfm]: MP3
[21:07:40] [Client thread/INFO] [openfm]: Starting Stream: http://listen.radionomy.com/DeepHouse at X:-2136 Y:4 Z:-1133
[21:07:40] [Thread-45/INFO] [openfm]: Device was null creating a new audio device via FactoryRegistry
[21:07:40] [Thread-45/INFO] [openfm]: Current volume values:
[21:07:40] [Thread-45/INFO] [openfm]: Player Volume: 0.0
[21:07:40] [Thread-45/INFO] [openfm]: Master Volume: 0.39072847
[21:07:40] [Thread-45/INFO] [openfm]: Note Block / Jukebox volume: 1.0
[21:07:40] [Thread-45/INFO] [openfm]: Result: 0.0
[21:07:40] [Thread-45/INFO] [openfm]: Listener was not null, calling playbackStarted
[21:07:45] [Client thread/INFO] [openfm]: stop() was called
[21:07:45] [Client thread/INFO] [openfm]: close() was called
[21:07:45] [Client thread/INFO] [openfm]: out was not null, we're closing the output stream
[21:07:45] [Thread-45/INFO] [openfm]: out was null in decodeFrame()... this is bad.
[21:07:45] [Thread-45/INFO] [openfm]: Audio Out was null exiting decodeFrame()
[21:07:45] [Thread-45/INFO] [openfm]: We're out of frames!
[21:07:45] [Thread-45/INFO] [openfm]: AudioDevice was null aborting

**Comment by CaitlynMainer** [comment]
Hmm... how long did you leave it running for? just the 5ish seconds in the log?  Also, could you get me a screenshot of your radio GUI?

**Comment by TTimoNN** [comment]
http://i.imgur.com/E0vhdWe.png
Left it on for 30 seconds and counting, no changes in the log.

**Comment by CaitlynMainer** [comment]
it's odd... it's reading the radio's volume as 0.0, though you've got it set to 9...  Also, looking at the log, 5 seconds after it started something called stop and everything flushed THEN it tried to play the stream.

**Comment by TTimoNN** [comment]
I might've stopped and paused it within those 5 seconds, from your comment I assume that this is all related to the radio volume being 0

**Comment by CaitlynMainer** [comment]
I'm going to run another debug build, and this one is going to be VERY spammy you'll likely need to pastebin the output

**Comment by TTimoNN** [comment]
Did some tests and changed the player volume to something different before and after starting the stream, it keeps outputting player volume 0

**Comment by CaitlynMainer** [comment]
http://puu.sh/u4J56/82398451f7.jar

**Comment by TTimoNN** [comment]
800.000 lines of "sampler loop: 0.25"

**Comment by CaitlynMainer** [comment]
well.. that's something, I know the decoder is running... but why is it not playing.  *THIS* is why I wish I had a system it did this on...

**Comment by TTimoNN** [comment]
You're welcome to teamviewer into my rig if you have time.

**Comment by TTimoNN** [comment]
[21:35:05] [Thread-43/INFO] [openfm]: sampler loop ended
[21:35:05] [Thread-43/INFO] [openfm]: out is not null in decodeFrame

every now and then through the sampler loops

**Comment by CaitlynMainer** [comment]
Ok, this one should be a little less spammy, http://puu.sh/u4LiT/be6d1796bf.jar  the decoder is running but it's not outputting, and I can't understand why..

**Comment by TTimoNN** [comment]
http://termbin.com/9if4
I see you forced the player volume to 1

**Comment by CaitlynMainer** [comment]
yeah I figured full volume would rule out any issues with the volume getting floored in the sampler loop. so the device is not null, and is open, but no sound.. :/  I'm at a loss here.  I'll keep poking around at it

**Comment by CaitlynMainer** [comment]
This is a very simplified version of the writer: http://puu.sh/u4Mxh/713d8885ba.jar  if this works I think I know where the error is. (Warning it may be loud I'm not sure)

**Comment by TTimoNN** [comment]
`[22:26:32] [Client thread/INFO]: [CHAT] [OpenFM] Selected Speaker
[22:26:32] [Client thread/INFO]: [CHAT] [OpenFM] Speaker Added
[22:26:34] [Client thread/INFO] [inventorytweaks]: pcl.OpenFM.GUI.GuiRadio
[22:26:34] [Client thread/INFO] [inventorytweaks]: pcl.OpenFM.TileEntity.RadioContainer
[22:26:39] [Client thread/INFO] [openfm]: MP3
[22:26:39] [Client thread/INFO] [openfm]: Starting Stream: http://listen.radionomy.com/DeepHouse at X:1554 Y:4 Z:-480
[22:26:39] [Thread-43/INFO] [openfm]: Device was null creating a new audio device via FactoryRegistry
[22:26:39] [Thread-43/INFO] [openfm]: Current volume values:
[22:26:39] [Thread-43/INFO] [openfm]: Player Volume: 1.0
[22:26:39] [Thread-43/INFO] [openfm]: Master Volume: 1.0
[22:26:39] [Thread-43/INFO] [openfm]: Note Block / Jukebox volume: 1.0
[22:26:39] [Thread-43/INFO] [openfm]: Result: 1.0
[22:26:39] [Thread-43/INFO] [openfm]: Listener was not null, calling playbackStarted
[22:26:56] [Client thread/INFO] [openfm]: stop() was called: stop
[22:26:56] [Client thread/INFO] [openfm]: close() was called by: stop
[22:26:56] [Client thread/INFO] [openfm]: out was not null, we're closing the output stream
[22:26:56] [Thread-43/INFO] [openfm]: out was null in decodeFrame()... this is bad.
[22:26:56] [Thread-43/INFO] [openfm]: Audio Out was null exiting decodeFrame()
[22:26:56] [Thread-43/INFO] [openfm]: We're out of frames!
[22:26:56] [Thread-43/INFO] [openfm]: AudioDevice was null aborting
[22:26:57] [Client thread/INFO] [openfm]: MP3
[22:26:57] [Client thread/INFO] [openfm]: Starting Stream: http://listen.radionomy.com/DeepHouse at X:1554 Y:4 Z:-480
[22:26:57] [Thread-45/INFO] [openfm]: Device was null creating a new audio device via FactoryRegistry
[22:26:57] [Thread-45/INFO] [openfm]: Current volume values:
[22:26:57] [Thread-45/INFO] [openfm]: Player Volume: 1.0
[22:26:57] [Thread-45/INFO] [openfm]: Master Volume: 1.0
[22:26:57] [Thread-45/INFO] [openfm]: Note Block / Jukebox volume: 1.0
[22:26:57] [Thread-45/INFO] [openfm]: Result: 1.0
[22:26:57] [Thread-45/INFO] [openfm]: Listener was not null, calling playbackStarted`

Only Minecraft music :(

**Comment by CaitlynMainer** [comment]
can you pastebin the output of /etc/java-8-openjdk/sound.properties ?

**Comment by TTimoNN** [comment]
http://termbin.com/1lkb

**Comment by TTimoNN** [comment]
I tried some random java audio test program: http://pastebin.com/raw/JdzJerYB I got this

**Comment by CaitlynMainer** [comment]
`javax.sound.sampled.LineUnavailableException` "This situation arises most commonly when a requested line is already in use by another application."

**Comment by CaitlynMainer** [comment]
can you have something like a youtube video running with audio while MC is open?

**Comment by TTimoNN** [comment]
Yes, my guess it that that java test program is not using pulse but only alsa, alsa has trouble with multiple audio streams.

**Comment by CaitlynMainer** [comment]
Well the audio device is set in the config I had you check, there is nothing expressly set in yours.. I'm wondering if java is trying to use ALSA for everything which is why my streams can't run cause MC is using the device already.

**Comment by TTimoNN** [comment]
There's no way that MC is using alsa seeing as pulseaudio is already running on alsa.

**Comment by CaitlynMainer** [comment]
this is my sound.properties http://puu.sh/u4Ot6/b140b94246.txt  Lizzy's config doesn't have anything explicitly set like yours, but with the error from this other program It's looking like this is an issue with Alsa/pulse/java that only happens to some people.

**Comment by TTimoNN** [comment]
I tried your settings, no success.

**Comment by gamax92** [comment]
pulseaudio will create a virtual alsa device and set that as the default, and then output to specific other alsa devices. If you have pulseaudio running then even if multiple applications try to use alsa at the same time it'll work as it's all still routed through pulseaudio, so it's likely not to be caused by something trying to use alsa.

**Comment by CaitlynMainer** [comment]
@gamax92 just rewrote OpenFM to output via OpenAL, meaning audio should output from MC's sound system directly, please test this build: http://puu.sh/u7xpF/3548e1557e.jar and let me know if it works.

**Comment by TTimoNN** [comment]
Receiving all MC sounds but no stream audio from OpenFM, nothing in the log besides.
"[13:01:59] [Client thread/INFO] [openfm]: Starting Stream: http://listen.radionomy.com/DeepHouse at X:471 Y:4 Z:-1312"
